src/main/java/stsjorbsmod/cards/wanderer/Patron.java

```diff
@@ -0,0 +1,84 @@
+package stsjorbsmod.cards.wanderer;
+
+import com.megacrit.cardcrawl.actions.common.DamageAction;
+import com.megacrit.cardcrawl.cards.AbstractCard;
+import com.megacrit.cardcrawl.cards.DamageInfo;
+import com.megacrit.cardcrawl.characters.AbstractPlayer;
+import com.megacrit.cardcrawl.dungeons.AbstractDungeon;
+import com.megacrit.cardcrawl.monsters.AbstractMonster;
+import stsjorbsmod.JorbsMod;
+import stsjorbsmod.actions.DowngradeCardAction;
+import stsjorbsmod.actions.DestroyCardAction;
+import stsjorbsmod.cards.CustomJorbsModCard;
+import stsjorbsmod.cards.DowngradeableCard;
+import stsjorbsmod.characters.Wanderer;
+import stsjorbsmod.patches.EphemeralField;
+import stsjorbsmod.patches.SelfExhumeFields;
+
+import java.util.Optional;
+
+import static com.megacrit.cardcrawl.core.CardCrawlGame.languagePack;
+
+public class Patron extends CustomJorbsModCard implements DowngradeableCard {
+    public static final String ID = JorbsMod.makeID(Patron.class);
+
+    private static final CardRarity RARITY = CardRarity.UNCOMMON;
+    private static final CardTarget TARGET = CardTarget.ENEMY;
+    private static final CardType TYPE = CardType.ATTACK;
+    public static final CardColor COLOR = Wanderer.Enums.WANDERER_CARD_COLOR;
+
+    private static final int COST = 2;
+    private static final int DAMAGE = 40;
+    private static final int UPGRADE_PLUS_DAMAGE = 10;
+
+    public Patron() {
+        super(ID, COST, TYPE, COLOR, RARITY, TARGET);
+        baseDamage = DAMAGE;
+        EphemeralField.ephemeral.set(this, true);
+        SelfExhumeFields.selfExhumeOnSnap.set(this, true);
+        purgeOnUse = true;
+    }
+
+    @Override
+    public void use(AbstractPlayer p, AbstractMonster m) {
+        addToBot(new DamageAction(m, new DamageInfo(p, damage)));
+
+        Optional<AbstractCard> cardOp = AbstractDungeon.player.masterDeck.group.stream().filter(c -> c.uuid == uuid).findAny();
+        if(!cardOp.isPresent()) {
+            JorbsMod.logger.info("Failed to purge a card we didn't have. Perhaps Duplication Potion or Attack Potion or similar effect occurred.");
+            return;
+        }
+        // In the highly unlikely event that UUIDs clash and the found card isn't actually the specific Patron card... oops
+        AbstractCard masterCard = cardOp.get();
+        if (upgraded) {
```

Comment:
For both downgrade and upgrade effects, I'd recommend passing the actions UUIDs rather than cards instances and using those to search for all in-battle instances, as well (this should resolve the issue you mention in the PR description):

```java
masterCopy = StSLib.getMasterDeckEquivalent(originalCard)
allInstances = GetAllInBattleInstances.get(originalCard.uuid);
if (masterCopy != null) {
    allInstances.add(masterCopy);
}
for (AbstractCard c : allInstances) {
    // perform purge / downgrade effect
}
```

Reply:
oh didn't realized that was a thing. seems good. though I probably don't need the allInstances anymore.